src/super_gradients/common/sg_loggers/wandb_sg_logger.py

```diff
@@ -246,7 +253,7 @@ def add_checkpoint(self, tag: str, state_dict: dict, global_step: int = 0):
         if self.save_checkpoints_wandb:
             if self.s3_location_available:
                 self.model_checkpoints_data_interface.save_remote_checkpoints_file(self.experiment_name, self._local_dir, name)
-            wandb.save(glob_str=path, base_path=self._local_dir, policy="now")
+            save_artifact(path)
```

Comment:
will this have the same effect? 
currently all files are saved inside the wandb experiment. Will the artifact be saved in the same way?

Reply:
Hi @ofrimasad 
You can refer to https://github.com/Deci-AI/super-gradients/pull/1010#issuecomment-1561244431 regarding this.

Reply:
After testing this, I can see two issues with this change:

1. it dramatically inflates the storage required by the user. Instead of saving only the last version of the model, it saves *all* the checkpoints. Users that are using an on-prem wandb server (like we do) will see a huge increase in storage usage.  

2. you are uploading different files to the same Artifact. During the training, we save ckpt_best.pth, ckpt_latest.pth, ckpt_average.pth. and this code maps all of them to the same Artifact (as different versions)  /

Please add a parameter (in the init), so the users can choose if they want the files to be saved as Artifact or not. 
please differentiate between files with different names.